nncf/common/tensor_numpy.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple, TypeVar, Union
+
+import numpy as np
+
+from nncf.common.tensor import NNCFTensorExt
+
+TensorType = TypeVar("TensorType")
+DeviceType = TypeVar("DeviceType")
+
+
+class NUMPYNNCFTensor(NNCFTensorExt):
```

Comment:
The major comment form my side:
- nncf.Tensor should be like a torch.Tensor or tf.Tensor.
- nncf.Tensor should be provided semiless integration torch.Tensor, tf.Tensor and numpy array.
- nncf.Tensor should be easily extended to another data representation.
- math functions should be provided for nncf.Tensor

I would recommend to think how to encapsulate all logic work with different data representation into nncf.Tensor abstraction.